crapssim/bet.py

```diff
@@ -177,48 +186,93 @@ def allowed(self, player) -> bool:
 class BaseOdds(WinningLosingNumbersBet, ABC):
     @property
     @abstractmethod
-    def base_bet_type(self) -> type:
+    def base_bet_types(self) -> tuple[typing.Type[WinningLosingNumbersBet]]:
         pass
 
-    def allowed(self, player) -> bool:
-        max_bet = self.bet_amount
-
-
+    @property
+    @abstractmethod
+    def table_odds_setting(self) -> str:
+        pass
 
-class Odds(WinningLosingNumbersBet):
-    def __init__(self, bet_amount: typing.SupportsFloat, bet_object: PassLine | Come):
-        super().__init__(bet_amount, None)
-        self.bet_object = bet_object
+    @property
+    @abstractmethod
+    def winning_number(self) -> int:
+        pass

```

Comment:
What's the reasoning behind having a separate winning_number property in the BaseOdds bet?  Is it just so that there are clean class definitions like this?

```python
class Odds4(Odds):
     winning_number: int = 4
     payout_ratio: float = 2 / 1
```

Other bets, such as Place, have a similar setup, and I'm wondering if it's worth creating a separate abstract class like SingleNumberBet that inherits from WinningLosingNumbersBet?

Reply:
I added three new classes with [8a43294](https://github.com/skent259/crapssim/pull/20/commits/8a4329445698e61ebd1c6fedf2a2579fb70a7998).

SingleWinningNumberBet for if there is only ever one number that can be a winner. SingleLosingNumberBet which is for if only one number is ever the loser. These two do whats described above for SingleNumberBet. I split it into two so that bets like Any7 can still be a SingleWinningNumberBet even though there are multiple numbers that can cause them to lose.

I also added OneRollBet for bets where if the number isn't in the winning_numbers it is automatically in the losing_numbers. This helped simplify the one roll bets, and will allow get_bets to search them by the type OneRollBet.